rdr_service/genomic/genomic_job_components.py

```diff
@@ -1095,22 +1107,99 @@ def _get_new_biobank_samples(self, from_date):
                 )
             AND ss.test IN ("1ED04", "1SAL2")
             AND ss.rdr_created > :from_date_param
-            AND ps.consent_for_study_enrollment_time > :consent_cutoff_param
+            AND ps.consent_cohort = :cohort_3_param
         """
         params = {
             "sample_status_param": SampleStatus.RECEIVED.__int__(),
             "dob_param": GENOMIC_VALID_AGE,
             "general_consent_param": QuestionnaireStatus.SUBMITTED.__int__(),
-            "consent_cutoff_param": GENOMIC_VALID_CONSENT_CUTOFF.strftime("%Y-%m-%d"),
             "ai_param": Race.AMERICAN_INDIAN_OR_ALASKA_NATIVE.__int__(),
             "from_date_param": from_date.strftime("%Y-%m-%d"),
             "withdrawal_param": WithdrawalStatus.NOT_WITHDRAWN.__int__(),
             "suspension_param": SuspensionStatus.NOT_SUSPENDED.__int__(),
+            "cohort_3_param": ParticipantCohort.COHORT_CURRENT.__int__(),
         }
         with self.samples_dao.session() as session:
             result = session.execute(_new_samples_sql, params).fetchall()
         return list(zip(*result))
 
+    def _get_new_c2_consent_samples(self, from_date):
+        """
+        Returns cohort 2 samples th
+        :param from_date:
+        :return:
+        """
+
+        _c2_samples_sql = """
+                SELECT DISTINCT
+                  ss.biobank_id,
+                  p.participant_id,
+                  o.biobank_order_id,
+                  o.collected_site_id,
+                  ss.biobank_stored_sample_id,
```

Comment:
Why did you go with DISTINCT instead of using these fields in a GROUP BY clause?

Reply:
In this case, I only needed unique rows, I wasn't aggregating the results, so I decided to use DISTINCT. For this query, DISTINCT slightly outperforms GROUP BY. I did some comparison testing with the large dataset on PROD and DISTINCT seemed to beat GROUP BY on average between 15-20%.

Reply:
Sounds good, thanks for the info.

Reply:
No problem, thanks for keeping me honest!